bindings/java/src/main/com/apple/foundationdb/FDBTenant.java

```diff
@@ -202,15 +200,12 @@ public CompletableFuture<Long> verifyBlobRange(byte[] beginKey, byte[] endKey, l
 
 	@Override
 	public CompletableFuture<Long> getId(Executor e) {
-		if (this.idFuture == null) {
-			pointerReadLock.lock();
-			try {
-				this.idFuture = new FutureInt64(Tenant_getId(getPtr()), e);
-			} finally {
-				pointerReadLock.unlock();
-			}
+		pointerReadLock.lock();
```

Comment:
I think we should keep the optimization for Java where it is easily implemented with the existing lock. We should only remove it for Python for now.